src/hdmf/data_utils.py

```diff
@@ -12,6 +12,92 @@
 from .utils import docval, getargs, popargs, docval_macro, get_data_shape
 
 
+def find_nth_none(lst, n):
```

Comment:
Is this function intended for external use? If we will only use this for ``array_with_desired_product`` method, then I would suggest to make the method private.
```suggestion
def __find_nth_none(lst, n):
```

Reply:
You're right, both of these should be private, though @CodyCBakerPhD 's solution makes this function unnecessary.